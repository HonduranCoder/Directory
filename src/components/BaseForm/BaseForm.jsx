export default function BaseForm({
  handleSubmit,
  email,
  label,
  setEmail,
  password,
  setPassword,
}) {
  return (
    <form onSubmit={handleSubmit}>
      <section>
        <label>
          Email
          <input
            id="email"
            type="email"
            name="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
        </label>
      </section>
      <section>
        <label>
          Password
          <input
            id="password"
            type="password"
            name="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
        </label>
      </section>
      <button type="submit">{label}</button>
    </form>
  );
}
