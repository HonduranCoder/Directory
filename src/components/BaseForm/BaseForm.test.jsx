import React from 'react';
import { render } from '@testing-library/react';
import BaseForm from './BaseForm.jsx';

it('should render form', () => {
  const { container } = render(<BaseForm />);
  expect(container).toMatchSnapshot();
});
