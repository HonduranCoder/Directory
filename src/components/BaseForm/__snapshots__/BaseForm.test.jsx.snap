// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render form 1`] = `
<div>
  <form>
    <section>
      <label>
        Email
        <input
          id="email"
          name="email"
          type="email"
          value=""
        />
      </label>
    </section>
    <section>
      <label>
        Password
        <input
          id="password"
          name="password"
          type="password"
          value=""
        />
      </label>
    </section>
    <button
      type="submit"
    />
  </form>
</div>
`;
